boot/bootstate20.go

```diff
@@ -283,10 +282,10 @@ func (ks20 *bootState20Kernel) markSuccessful(update bootStateUpdate) (bootState
 		// On commit, set CurrentKernels as just this kernel because that is the
 		// successful kernel we booted
 		u20.writeModeenv.CurrentKernels = []string{sn.Filename()}
-	}
 
-	// keep track of the device for resealing
-	u20.setDevice(ks20.dev)
+		// keep track of the model for resealing
+		u20.resealForModel(ks20.dev.Model())
```

Comment:
mmm good catch moving this inside the if, eventually we should probably just try and get rid of this if to be extra safe, more test code is probably okay long term